src/ast/nodes/Identifier.ts

```diff
@@ -158,22 +159,32 @@ export default class Identifier extends NodeBase {
 		}
 	}
 
-	render(code: MagicString, options: RenderOptions) {
+	render(
+		code: MagicString,
+		options: RenderOptions,
+		{ renderedParent, fieldOfRenderedParent }: NodeRenderOptions = BLANK
+	) {
 		if (this.variable) {
 			const name = this.variable.getName();
 
 			if (name !== this.name) {
 				code.overwrite(this.start, this.end, name, {
 					storeName: true,
-					contentOnly: false
+					contentOnly: true
 				});
-
-				// special case
 				if (this.parent.type === NodeType.Property && (<Property>this.parent).shorthand) {
-					code.appendLeft(this.start, `${this.name}: `);
+					code.prependRight(this.start, `${this.name}: `);
 				}
 			}
-
+			// In strict mode, any variable named "eval" must be the actual "eval" function
+			if (
+				name === 'eval' &&
+				renderedParent &&
+				renderedParent.type === 'CallExpression' &&
+				fieldOfRenderedParent === 'callee'
```

Comment:
Nice! Slightly shorter names might be `renderParent` and `renderParentField`, but perhaps these are more descriptive anyway.